bundles/test/org.eclipse.smarthome.magic/src/main/java/org/eclipse/smarthome/magic/binding/internal/MagicServiceImpl.java

```diff
@@ -49,4 +59,25 @@
         return null;
     }
 
+    @Activate
+    public void activate(Map<String, Object> properties) {
+        modified(properties);
+    }
+
+    @Modified
+    public void modified(Map<String, Object> properties) {
+        // Print changed values
+        MagicServiceConfig config = new Configuration(properties).as(MagicServiceConfig.class);
+        List<Field> l = ConfigUtil.getAllFields(MagicServiceConfig.class);
```

Comment:
Maybe instead of exposing the reflection based method `getAllFields` you could give a `toString` to MagicServiceConfig.

Reply:
True. Regarding the annotations: I actually introducted `UseKey` because the configuration key is unfortunately called "boolean".

I'm honestly a big fan of this property-map to class object mapping, partly because I do not find that powerful reflection API in many other languages. However I'd like to see this as a default way of handling service configuration and it should work in every case.

So we either need this `UseKey` annotation or we need to forbid certain configuration key names (Java language keywords).